### ISSUE TITLE
Restrict scopes in dynamic registration request for clients with password grant

### ISSUE THREAD

**Issue by yurem** [GluuFederation/oxAuth#1130]
Create a new oxAuth JSON property called `allowedPasswordGrantScopes`--a client dynamically registering for password grant can obtain only these scopes.

**Comment by nynymike** [comment]
I'm ok with a client calling `/Userinfo` with an access token obtained from ROPC. A Gluu Server admin can always manutally add scopes for a client. But a few issues I think we should address: 
* We should move default ROPC scopes obtained during dynamic clietn registration to a system level JSON property. 
* There should be no default ROPC scopes, including `openid` 
* To calculate `sub` for pairwise id, use the `username` claim from the access token

**Comment by yurem** [comment]
Instead of adding new oxAuth property we can add nw attribute to scope entry to specify the same. Example:
```
dn: inum=6D90,ou=scopes,o=gluu
...
defaultScope: true
defaultPassowrdGrantScope: true
...
```

For me this will be better because we don't need to update property in oxAuth config after scope removal, for example.
@nynymike what do you think?

**Comment by nynymike** [comment]
Yes, agreed. This would allow us to control the behavior on a scope-by-scope basis.

**Comment by yuriyz** [comment]
I will take over.

**Comment by yuriyz** [comment]
Implemented in 4.2.1 and master.
